CppCoreGuidelines.md

```diff
@@ -13138,12 +13138,21 @@ C++ provides better type checking and more notational support.
 It provides better support for high-level programming and often generates faster code.
 
 ##### Example
-
-    char ch = 7;
-    void* pv = &ch;
-    int* pi = pv;   // not C++
-    *pi = 999;      // overwrite sizeof(int) bytes near &ch
-
+    struct dataX{
+        char temperature;
+        char windspeed;
+    }
+    struct dataY{
+        int id;
+        double weight;
+    }
+    int main(){
+        struct dataX x;
+        void* ptrX = &x;
+        struct dataY* = ptrX; // not C++
+        dataY.temperature=20;// overwrite sizeof(int) bytes near &temperature, 
+        //thus corrupting "windspeed" and other data or structure padding
```

Comment:
In C void* is compatible with any non-const pointer to structure.

Your C code invokes undefined behaviour as it is in violation of the aliasing rule (incompatible types).